### PR TITLE
Updated jax2tf README.md

### PR DESCRIPTION
**Pull request by krvaibhaw**
Updated jax2tf README.md

### CODE REVIEW

**Comment by krvaibhaw** [comment]
@googlebot I signed it!

**Comment by jakevdp** [comment]
Thanks! I think some of these changes may conflict with https://github.com/google/jax/pull/6941, which was just merged this morning. Can you rebase your PR on the current master branch?

File: setup.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
     package_data={'jax': ['py.typed']},
     python_requires='>=3.6',
     install_requires=[
-        'numpy>=1.17',
+        'numpy >=1.17',
```
Comment (hawkinsp) [inline setup.py:37]: Revert this change.

File: build/test-requirements.txt [hunk 2]
```diff
@@ -2,10 +2,10 @@ flake8
 flatbuffers==1.12
 # For now, we pin the numpy version here
 numpy>=1.16
-mypy==0.790
+mypy==0.902
```
Comment (hawkinsp) [inline build/test-requirements.txt:5]: Please do not mix different kinds of changes in the same PR. Bumping the mypy version and improving the docs are both fine things to do, but please do them in separate PRs that try to do one thing only.

**Review by hawkinsp — CHANGES_REQUESTED** [review]
I'd split this PR up into two, and give each a better title (e.g., I'm not sure how this is related to security, for example...)

File: build/test-requirements.txt (hunk 2 above)
Comment (hawkinsp) [reply build/test-requirements.txt:5]: This still need to be addressed. I would just revert the mypy change from this PR and send it as a separate PR.

Please also squash your commits and fix the title and commit comments to reflect that this is a doc-only change.

File: jax/experimental/jax2tf/README.md [hunk 3]
```diff
@@ -305,11 +305,11 @@ dimension variables JAX computes intermediate shapes as multi-variate polynomial
 involving dimension variables, which are assumed to range over strictly positive
 integers.
 The dimension polynomials have the following behavior for arithmetic operations:
-  * addition, subtraction, multiplication are supported without restrictions, and
+  * addition, subtraction, multiplication are supported without restrictions and
```
Comment (hawkinsp) [inline jax/experimental/jax2tf/README.md:308]: Revert this change. I think the comma makes it read more clearly.

File: jax/experimental/jax2tf/README.md [hunk 4]
```diff
@@ -305,11 +305,11 @@ dimension variables JAX computes intermediate shapes as multi-variate polynomial
...
     are overloaded, such that `+`, `*`, `np.sum`, `np.prod` work directly on
     dimension polynomials.
     These arise, e.g., in `jax.numpy.concatenate` or `jax.numpy.reshape`.
-  * division is a special case. It is also overloaded, but it is only partially
+  * the division is a special case. It is also overloaded, but it is only partially
```
Comment (hawkinsp) [inline jax/experimental/jax2tf/README.md:312]: Revert this change. It reads better without the "the".

File: jax/experimental/jax2tf/README.md [hunk 5]
```diff
@@ -701,7 +701,7 @@ We use the following TFXLA ops:
    * `XlaConv` and `XlaConv2` (wrap XLA ConvGeneralDilated operator).
    * `XlaDot` and `XlaDotV2` (wrap XLA DotGeneral operator).
    * `XlaGather` (wraps XLA Gather operator). We could use `tf.gather` in some
-     cases but not always. Also, `tf.gather` has a different semantics than `lax.gather`
+     cases but not always. Also, `tf.gather` has different semantics than `lax.gather`
```
Comment (hawkinsp) [inline jax/experimental/jax2tf/README.md:704]: It's not clear to me this change is a correct usage of "semantics". Revert it.

File: jax/experimental/jax2tf/README.md [hunk 6]
```diff
@@ -738,18 +738,18 @@ result in different results than JAX’s PRNG.
...
 if one were to write the same functionality in JAX idiomatic code vs.
 native TensorFlow idiomatic code we could end up with very different compilation paths.
-Take for example, the case of batch normalization.
+Take, for example, the case of batch normalization.
 In TensorFlow if one uses [tf.nn.batch_normalization](https://www.tensorflow.org/api_docs/python/tf/nn/batch_normalization),
 a “high-level” TensorFlow op for batch
 normalization is generated, and in the absence of XLA, on CPU or GPU,
 a custom C++ “high-level” kernel implementing batch normalization is executed.
-In JAX, there is no primitive for batch normalization, and instead the
+In JAX, there is no primitive for batch normalization, and instead, the
```
Comment (hawkinsp) [inline jax/experimental/jax2tf/README.md:746]: Revert this change. It reads better without the additional comma.

File: jax/experimental/jax2tf/README.md [hunk 7]
```diff
@@ -738,18 +738,18 @@ result in different results than JAX’s PRNG.
...
 operation is decomposed into low-level primitives (e.g., [flax.nn.BatchNorm](https://flax.readthedocs.io/en/latest/_autosummary/flax.nn.BatchNorm.html#flax.nn.BatchNorm),
 or haiku.BatchNorm).
 Once those primitives are converted to TensorFlow, and the resulting code is
 run without XLA, the ensemble of the kernels executed will quite
 possibly behave differently, performance-wise or even numerically,
-than either the TensorFlow native or JAX native batch normalization.
+then either the TensorFlow native or JAX native batch normalization.
```
Comment (hawkinsp) [inline jax/experimental/jax2tf/README.md:752]: Revert this change. "differently than" is the correct spelling.

**Review by hawkinsp — CHANGES_REQUESTED** [review]

File: build/test-requirements.txt [hunk 8]
```diff
@@ -8,4 +8,4 @@ pytest-benchmark
 pytest-xdist
 wheel
 # Install jax from the current directory; minimum required jaxlib from pypi.
-.[minimum-jaxlib]
\ No newline at end of file
+.[minimum-jaxlib]
```
Comment (hawkinsp) [inline build/test-requirements.txt:11]: There's still an unnecessary change to this file.

**Review by hawkinsp — CHANGES_REQUESTED** [review]
Please squash your commits as well.

**Pull request closed without merging**